hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/CompactUtil.java

```diff
@@ -32,16 +32,16 @@ private CompactUtil() {
     static HazelcastSerializationException exceptionForUnexpectedNullValue(@Nonnull String fieldName,
                                                                            @Nonnull String methodSuffix) {
         return new HazelcastSerializationException("Error while reading " + fieldName + ". "
```

Comment:
These methods are used in compact generic records as well. So, we would need both versions and use the correct versions depending on the context (generic record vs compact reader)

Reply:
What about adding a boolean field to CompactInternalGenericRecord which is set to true if it is a DefaultCompactReader? If the boolean field is true we will print `read` otherwise it will be `get`

Reply:
But I guess we return DefaultCompactReader for generic record case, too. 😕

Reply:
I tried to conditionally print the correct methodPrefix. @mdumandag Can you check again?

Reply:
Instead of a boolean, I believe you can add a new parameter `@Nonnull String methodPrefix`. Then, introduce a method in CompactInternalGenericRecord, something like `getMethodPrefixForErrorMessages`, and override it in `DefaultCompactReader`. IMHO, this will be more clean and easier to read

Reply:
Done, for DeserializedGenericRecord I used a static constant variable. I also added some tests.